pyccl/tests/test_ept_power.py

```diff
@@ -0,0 +1,260 @@
+import numpy as np
+import pyccl as ccl
+import pytest
+
+
+COSMO = ccl.Cosmology(
+    Omega_c=0.27, Omega_b=0.045, h=0.67, sigma8=0.8, n_s=0.96,
+    transfer_function='bbks', matter_power_spectrum='linear')
+TRS = {'TG': ccl.nl_pt.PTNumberCountsTracer(b1=2.0, b2=2.0, bs=2.0),
+       'TI': ccl.nl_pt.PTIntrinsicAlignmentTracer(c1=2.0, c2=2.0,
+                                                  cdelta=2.0),
+       'TM': ccl.nl_pt.PTMatterTracer()}
+PTC = ccl.nl_pt.EulerianPTCalculator(with_NC=True, with_IA=True,
+                                     with_matter_1loop=True,
+                                     cosmo=COSMO)
+
+
+def test_ept_calculator_smoke():
+    c = ccl.nl_pt.EulerianPTCalculator(log10k_min=-3,
+                                       log10k_max=1,
+                                       nk_per_decade=10,
+                                       extra_params={'pad_factor': 2})
+    assert len(c.k_s) == 40
+
+
+@pytest.mark.parametrize('options', [['TG', 'TG', False, False],
+                                     ['TG', 'TG', False, True],
+                                     ['TG', 'TI', False, False],
+                                     ['TG', 'TM', False, False],
+                                     ['TI', 'TG', False, False],
+                                     ['TI', 'TI', False, False],
+                                     ['TI', 'TI', True, False],
+                                     ['TI', 'TM', False, False],
+                                     ['TM', 'TG', False, False],
+                                     ['TM', 'TI', False, False],
+                                     ['TM', 'TM', False, False]])
+def test_ept_get_pk2d_smoke(options):
+    if options[0] == options[1]:
+        t2 = None
+    else:
+        t2 = TRS[options[1]]
```

Comment:
For example, this can be really neatly put into a one-liner, which is easy to read. I don't know why you are so against them: `t2 = TRS[options[1]] if options[0] != options[1] else None`.

Reply:
Subjective, coding style